### ISSUE TITLE
Handling winit versions/semver breaking changes

### ISSUE THREAD

**Issue by aclysma** [imgui-rs/imgui-rs#412]
I recently had one of my crates break due to a change in the winit-23 feature flag behavior:
 * imgui_winit_support 0.6.0 will use winit 0.23 when using winit-23 feature flag
 * imgui_winit_support 0.6.1 will use winit 0.24 when using winit-23 feature flag

Winit is being used in the public API, so I think this is technically a semver-breaking change. Would keeping the winit feature flags tied to an exact semver version of winit would be a way to avoid this? Bumping imgui_winit_support to 0.7.0 rather than 0.6.1 might have worked too.

I definitely understand juggling winit versions is not fun, and unfortunately I haven't found a better solution that using feature flags!

**Comment by thomcc** [comment]
Ugh, what an awful situation. I had hoped it would be constrained by what other dependencies are needed. It's likely that all the version range-based stuff we do in the winit-support crate is similarly invalid then.

I won't be able to fix this until after the new year, unfortunately.

**Comment by aclysma** [comment]
Yeah, I was hoping for that behavior too. In one of my other crates I went the route of feature flags for every winit semver version (based on this crate as an example actually). Thanks for looking, hope you have a nice break! :)
